src/coordinate_search.jl

```diff
@@ -0,0 +1,65 @@
+export coordinate_search
+
+"""
+	The coordinate search is a derivative free optimization method.
+	It minimizes a certain function f by walking on a grid and taking a step where the function values are smaller.
+	When it can no longer find any smaller values, it reduces the grid by a factor of 2, repeating the process.
+
+"""
+
+function coordinate_search(nlp :: AbstractNLPModel;
+			   tol = 1e-12,
+			   k = 0,
+			   α = 1.0,
+			   xk = nlp.meta.x0,
+			   el_time = 0.0,
+			   max_time :: Real  = 1e-1,
+			   max_eval = 500)
+
```

Comment:
Overall: don't use TABs, configure your editor to convert TABs to spaces.
Declare the types of the arguments.
`k` and `el_time` shouldn't be options.
`xk` should be only `x`.